src/wire/ipv6routing.rs

```diff
@@ -0,0 +1,664 @@
+        let len = self.buffer.as_ref().len();
+        if len < field::MIN_HEADER_SIZE {
+            return Err(Error::Truncated);
+        }
+
+        if len < field::DATA(self.header_len()).end as usize {
+            return Err(Error::Truncated);
+        }
+
+        Ok(())
+    }
+
+    /// Consume the header, returning the underlying buffer.
+    pub fn into_inner(self) -> T {
+        self.buffer
+    }
+
+    /// Return the next header field.
+    #[inline]
+    pub fn next_header(&self) -> Protocol {
+        let data = self.buffer.as_ref();
+        Protocol::from(data[field::NXT_HDR])
+    }
+
+    /// Return the header length field. Length of the Routing header in 8-octet units,
+    /// not including the first 8 octets.
+    #[inline]
+    pub fn header_len(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::LENGTH]
+    }
+
+    /// Return the routing type field.
+    #[inline]
+    pub fn routing_type(&self) -> Type {
+        let data = self.buffer.as_ref();
+        Type::from(data[field::TYPE])
+    }
+
+    /// Return the segments left field.
+    #[inline]
+    pub fn segments_left(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::SEG_LEFT]
+    }
+
+    /// Return the IPv6 Home Address
+    ///
+    /// # Panics
+    /// This function may panic if this header is not the Type2 Routing Header routing type.
+    pub fn home_address(&self) -> Address {
+        let data = self.buffer.as_ref();
+        Address::from_bytes(&data[field::HOME_ADDRESS])
+    }
+
+    /// Return the number of prefix octects elided from addresses[1..n-1].
+    ///
+    /// # Panics
+    /// This function may panic if this header is not the RPL Source Routing Header routing type.
+    pub fn cmpr_i(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::CMPR] >> 4
+    }
+
+    /// Return the number of prefix octects elided from the last address (addresses[n]).
+    ///
+    /// # Panics
+    /// This function may panic if this header is not the RPL Source Routing Header routing type.
+    pub fn cmpr_e(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::CMPR] & 0xf
+    }
+
+    /// Return the number of octects used for padding after addresses[n].
+    ///
+    /// # Panics
+    /// This function may panic if this header is not the RPL Source Routing Header routing type.
+    pub fn pad(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::PAD] >> 4
+    }
+
+    /// Return the address vector in bytes
+    ///
+    /// # Panics
+    /// This function may panic if this header is not the RPL Source Routing Header routing type.
+    pub fn addresses(&self) -> &[u8] {
+        let data = self.buffer.as_ref();
+        &data[field::ADDRESSES(data[field::LENGTH])]
+    }
+}
+
+impl<'a, T: AsRef<[u8]> + ?Sized> Header<&'a T> {
+    /// Return type specific data
+    #[inline]
+    pub fn data(&self) -> &'a[u8] {
```

Comment:
Is this really used? Type 2 uses `home_address` and RPL uses `address`.

Reply:
It was used in a test. I removed `data` and `data_mut` and modified the test.